packages/util/src/logger/cloud-sdk-logger.spec.ts

```diff
@@ -240,6 +241,125 @@ describe('Cloud SDK Logger', () => {
+        etag: 'W/"134d2-7mWFsXsjM/3aYrrtmSWe4HUcaGE"',
+        'expect-ct': 'max-age=0',
+        'origin-agent-cluster': '?1',
+        'permissions-policy': 'interest-cohort=()',
+        'referrer-policy': 'no-referrer',
+        server: 'nginx',
+        'strict-transport-security': 'max-age=15552000; includeSubDomains',
+        'x-content-type-options': 'nosniff',
+        'x-dns-prefetch-control': 'off',
+        'x-download-options': 'noopen',
+        'x-frame-options': 'SAMEORIGIN',
+        'x-permitted-cross-domain-policies': 'none',
+        'x-xss-protection': '0'
+      };
+      // Should return a copy of the input
+      expect(sanitizeRecord(responseHeaders)).not.toBe(responseHeaders);
+      expect(sanitizeRecord(responseHeaders)).toEqual(responseHeaders);
+    });
+
+    it('does not remove any unwanted record properties for common request headers', () => {
+      const requestHeaders = {
+        ':authority': 'example.com',
+        ':method': 'GET',
+        ':path': '/api/v1',
+        ':scheme': 'https',
+        accept: '*/*',
+        'accept-encoding': 'gzip, deflate, br',
+        'accept-language': 'en-US,en;q=0.9,de;q=0.8',
+        'content-type': 'application/json',
+        cookie: 'USERID=1;SERVERID=2',
+        dnt: '1',
+        'if-none-match': 'W/"134d2-7mWFsXsjM/3aYrrtmSWe4HUcaGE"',
+        referer: 'https://example.com',
+        'sec-ch-ua':
+          '"Not A;Brand";v="99", "Chromium";v="98", "Google Chrome";v="98"',
+        'sec-ch-ua-mobile': '?0',
+        'sec-ch-ua-platform': '"macOS"',
+        'sec-fetch-dest': 'empty',
+        'sec-fetch-mode': 'cors',
+        'sec-fetch-site': 'same-origin',
+        'user-agent':
+          'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.80 Safari/537.36'
+      };
+      // Should return a copy of the input
+      expect(sanitizeRecord(requestHeaders)).not.toBe(requestHeaders);
+      expect(sanitizeRecord(requestHeaders)).toEqual(requestHeaders);
+    });
+
+    it('should remove auth headers', () => {
+      const input = {
+        access_token: 'SECRET',
+        my_access_token: 'SECRET',
+        authentication: 'secret',
+        service_authentication: 'secret',
+        authorization: 'secret',
+        Authorization: 'secret',
+        AUTHORIZATION: 'secret',
+        credentials: 'password',
+        'x-xsrf-token': 'secret token',
+        cookie: 'csrfToken=secret'
+      };
+      const output = {
+        access_token: '<DATA NOT LOGGED TO PREVENT LEAKING SENSITIVE DATA>',
+        my_access_token: '<DATA NOT LOGGED TO PREVENT LEAKING SENSITIVE DATA>',
+        authentication: '<DATA NOT LOGGED TO PREVENT LEAKING SENSITIVE DATA>',
+        service_authentication:
+          '<DATA NOT LOGGED TO PREVENT LEAKING SENSITIVE DATA>',
+        authorization: '<DATA NOT LOGGED TO PREVENT LEAKING SENSITIVE DATA>',
+        Authorization: '<DATA NOT LOGGED TO PREVENT LEAKING SENSITIVE DATA>',
```

Comment:
Add an exception for the "credentials". Perhaps you can have a look if codeql has a value for white listed passwords or a way to add these value to our run.

Reply:
In the discussion with the GHAS people they gave arguments to run the scan also on test code. So I guess we have to manually exclude the false positive then.